lib/Analysis/Allocation.cpp

```diff
@@ -267,12 +233,9 @@ class AllocationAnalysis {
       } else {
         auto smemShape = getRepShapeForAtomic(op->getResult(0));
         auto elems = getNumScratchElements(smemShape);
-        auto elemTy =
-            cast<triton::PointerType>(value.getType()).getPointeeType();
+        auto elemTy = cast<PointerType>(value.getType()).getPointeeType();
         auto bytes =
-            isa<triton::PointerType>(elemTy)
-                ? elems * kPtrBitWidth / 8
-                : elems * std::max<int>(8, elemTy.getIntOrFloatBitWidth()) / 8;
+            elems * std::max<int>(8, elemTy.getIntOrFloatBitWidth()) / 8;
```

Comment:
should be `elems * kPtrBitWidth / 8`?

Reply:
What I meant is that the element type would not be a pointer type, so `isa` should always be false. Let me add an assert here maybe in case we run into any problems

Reply:
oh yeah you're absolutely right I read too fast